packages/next/auth.tsx

```diff
@@ -299,16 +307,24 @@ const AuthProvider: React.FC = ({ children }) => {
                     authenticatedUser: null,
                 },
             })
+            setAuthLoading(false)
         },
         onError: (error) => {
             console.error(error)
+            setAuthLoading(false)
         },
     })
 
+    useEffect(() => {
+        if (userLoading || signOutLoading || signInLoading) {
+            setAuthLoading(true)
+        }
+    }, [userLoading, signOutLoading, signInLoading])
+
     return (
         <AuthContext.Provider
             value={{
-                isLoading: userLoading || signOutLoading || signInLoading,
+                isLoading: userLoading || signOutLoading || signInLoading || isAuthLoading,
```

Comment:
`isAuthLoading` will be true if `userLoading || signOutLoading || signInLoading` is equals true. Do we really need `isAuthLoading`?